clippy_lints/src/bool_to_int_with_if.rs

```diff
@@ -55,27 +55,43 @@ fn check_if_else<'tcx>(ctx: &LateContext<'tcx>, expr: &'tcx rustc_hir::Expr<'tcx
     if let ExprKind::If(check, then, Some(else_)) = expr.kind
         && let Some(then_lit) = int_literal(then)
         && let Some(else_lit) = int_literal(else_)
-        && check_int_literal_equals_val(then_lit, 1)
-        && check_int_literal_equals_val(else_lit, 0)
     {
+        let inverted = if
+            check_int_literal_equals_val(then_lit, 1)
+            && check_int_literal_equals_val(else_lit, 0) {
+            false
+        } else if
+            check_int_literal_equals_val(then_lit, 0)
+            && check_int_literal_equals_val(else_lit, 1) {
+            true
+        } else {
+            // Expression isn't boolean, exit
+            return;
+        };
         let mut applicability = Applicability::MachineApplicable;
         let snippet = snippet_block_with_applicability(ctx, check.span, "..", None, &mut applicability);
+
+        let invert = if inverted { "!" } else { "" };
+        let need_parens = should_have_parentheses(check);
```

Comment:
Clippy has an helper struct that will already handle parenthesis for you, including when negating the expression: [clippy_utils](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/index.html)::[sugg](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/sugg/index.html)::[Sugg](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/sugg/enum.Sugg.html#)

Reply:
That fixed a bug in the lint, and in turn fixed a bug in `clippy_utils::sugg::Sugg`, so thanks!

[bug in `Sugg`](https://github.com/rust-lang/rust-clippy/pull/9476/commits/dd97c1ed205f47d13dc5bc482a73c0c76383d159)